test/common/router/config_impl_test.cc

```diff
@@ -2414,21 +2420,20 @@ TEST_F(RouterMatcherHashPolicyTest, HashHeaders) {
 TEST_F(RouterMatcherHashPolicyTest, HashHeadersRegexSubstitution) {
   // Apply a regex substitution before hashing.
   auto* header = firstRouteHashPolicy()->mutable_header();
-  header->set_header_name(":path");
+  header->set_header_name("foo_header");
   auto* regex_spec = header->mutable_regex_rewrite();
   regex_spec->set_substitution("\\1");
   auto* pattern = regex_spec->mutable_pattern();
   pattern->mutable_google_re2();
   pattern->set_regex("^/(\\w+)$");
   {
-    Http::TestRequestHeaderMapImpl headers = genHeaders("www.lyft.com", "/foo", "GET");
-    Router::RouteConstSharedPtr route = config().route(headers, 0);
-    const auto foo_hash_value = 3728699739546630719;
-    EXPECT_EQ(route->routeEntry()
-                  ->hashPolicy()
-                  ->generateHash(nullptr, headers, add_cookie_nop_, nullptr)
-                  .value(),
-              foo_hash_value);
+    EXPECT_FALSE(generateHash({}));
+    EXPECT_TRUE(generateHash({"/bar"}));
+
+    absl::Hash<const std::vector<absl::string_view>> hasher;
+    EXPECT_EQ(hasher({"bar", "foo"}), generateHash({"/bar", "/foo"}));
```

Comment:
see above

then we don't need to call the absl hasher directly in the test code.